app/src/main/java/com/hedvig/app/feature/marketing/data/GetMarketingBackgroundUseCase.kt

```diff
@@ -0,0 +1,17 @@
+package com.hedvig.app.feature.marketing.data
+
+import com.apollographql.apollo.ApolloClient
+import com.hedvig.android.owldroid.graphql.MarketingBackgroundQuery
+import com.hedvig.app.util.LocaleManager
+import com.hedvig.app.util.apollo.safeQuery
+
+class GetMarketingBackgroundUseCase(
+    private val apolloClient: ApolloClient,
+    private val localeManager: LocaleManager,
+) {
+    suspend operator fun invoke() = apolloClient
+        .query(MarketingBackgroundQuery(localeManager.defaultLocale().rawValue))
+        .safeQuery()
+        .toEither()
+        .map { it.appMarketingImages.firstOrNull() }
```

Comment:
We could map this to our own domain model, and perhaps return error if blurhash etc. are null? Then we can skip the safeLets in the viewmodel.